upstairs/src/dummy_downstairs_tests.rs

```diff
@@ -2885,6 +2885,97 @@ async fn test_no_send_offline() {
     }
 }
 
+#[tokio::test]
+async fn test_ro_activate_with_two() {
+    let log = csl();
+
+    let cfg = DownstairsConfig {
+        read_only: true,
+        reply_to_ping: true,
+        extent_count: DEFAULT_EXTENT_COUNT,
+        extent_size: Block::new_512(DEFAULT_BLOCK_COUNT),
+        gen_numbers: vec![0u64; DEFAULT_EXTENT_COUNT as usize],
+        flush_numbers: vec![0u64; DEFAULT_EXTENT_COUNT as usize],
+        dirty_bits: vec![false; DEFAULT_EXTENT_COUNT as usize],
+    };
+
+    let ds1 = cfg.clone().start(log.new(o!("downstairs" => 1))).await;
+    let ds2 = cfg.clone().start(log.new(o!("downstairs" => 2))).await;
+    let ds3 = cfg.clone().start(log.new(o!("downstairs" => 3))).await;
+
+    let (g, io) = Guest::new(Some(log.clone()));
+    let guest = Arc::new(g);
+
+    let crucible_opts = CrucibleOpts {
+        id: Uuid::new_v4(),
+        target: vec![ds1.local_addr, ds2.local_addr, ds3.local_addr],
+        flush_timeout: Some(1.0),
+        read_only: true,
+
+        ..Default::default()
+    };
+
+    let join_handle = up_main(crucible_opts, 1, None, io, None).unwrap();
+
+    let mut handles: Vec<JoinHandle<()>> = vec![];
+    {
+        let guest = guest.clone();
+        handles.push(tokio::spawn(async move {
+            guest.activate().await.unwrap();
+        }));
+    }
+
+    // Connect two Downstairs
+    //
+    // We're using a VecDeque instead of FuturesOrdered so that if a future
+    // panics, we get a meaningful backtrace with a correct line number.
+    let mut fut = VecDeque::new();
```

Comment:
Do we get anything from running these futures in a spawned task, versus doing it in order?
```
for mut ds in [ds1, ds2] {
    ds.negotiation_start().await;
    ds.negotiation_step_extent_versions_please().await;
}
```

Reply:
I don't actually know Mr. Chesterton.  I just copied the fence from another place where I found it doing a similar thing.

Reply:
Copied from: 
```
8da7be776 (Matt Keeter       2024-04-08 12:34:36 -0400  598)         // We're using a VecDeque instead of FuturesOrdered so that if a future 
```

I've also tested with just this:
```
    for ds in [&mut ds1, &mut ds2] {
        ds.negotiate_start().await;
        ds.negotiate_step_extent_versions_please().await;
    }
```
And, it works.  I did not see what happens on a panic though.

Reply:
Updated in 84ede33e